klippy/extras/probe_eddy_current.py

```diff
@@ -350,10 +350,11 @@ def pull_probed(self):
             if sensor_z <= -OUT_OF_RANGE or sensor_z >= OUT_OF_RANGE:
                 raise self._printer.command_error(
                     "probe_eddy_current sensor not in valid range")
-            # Callers expect position relative to z_offset, so recalculate
-            bed_deviation = toolhead_pos[2] - sensor_z
-            toolhead_pos[2] = self._z_offset + bed_deviation
-            results.append(toolhead_pos)
+            res = manual_probe.ProbeResult(
+                toolhead_pos[0]+self._offsets[0],
+                toolhead_pos[1]+self._offsets[1], toolhead_pos[2]-sensor_z,
+                toolhead_pos[0], toolhead_pos[1], toolhead_pos[2])
```

Comment:
AFAIU.
Caller generally set the toolhead position, and it seems to me that first 3 values is the position of nozzle over the bed (and so, second triplet is the probe position).

Anyway, with this code the probe output is changed and outputs around zero during `PROBE`/`PROBE_ACCURACY`, because normally bed_z pretty close to sensor_z.
So, with z_offset = 1,
toolhead_z = 1.0, sensor_z = 0.975
bed_z = 1.0 - 0.975 = 0.025

And swapping them, makes probe output looks as before (so +- z_offset).